src/main/java/it/gov/pagopa/payhub/auth/utils/JWTValidator.java

```diff
@@ -53,4 +56,31 @@ public Map<String, Claim> validate(String token, String urlJwkProvider) {
             throw new InvalidTokenException("The token is not valid");
         }
     }
+
+    /**
+     * Validates JWT signature with publickey.
+     *
+     * @param token the JWT to validate
+     * @param pubKey the public key (in PEM format) used to validate the token
+     * @throws IllegalStateException if the public key cannot be loaded due to
+     *         invalid format, missing algorithm, or I/O issues.
+     * @throws TokenExpiredException if the token has expired.
+     * @throws InvalidTokenException if the token is invalid for any other reason
+     *         (e.g., signature verification failure).
+     */
+    public void validateInternalToken(String token, String pubKey) {
+        RSAPublicKey rsaPublicKey = null;
+        try{
+            rsaPublicKey = CertUtils.pemPub2PublicKey(pubKey);
+            Algorithm algorithm = Algorithm.RSA512(rsaPublicKey);
+            JWTVerifier verifier = JWT.require(algorithm).build();
```

Comment:
The publicKey is a constant for the entire application life.
If it will change, the application should restart.

Thus, the entire verifier object could be created once at startup instead of to be created at each invocation